src/transporters/amqp.js

```diff
@@ -0,0 +1,329 @@
+/*
+ * moleculer
+ * Copyright (c) 2017 Ice Services (https://github.com/ice-services/moleculer)
+ * MIT Licensed
+ */
+
+"use strict";
+
+const Promise		= require("bluebird");
+const Transporter 	= require("./base");
+const {
+	PACKET_REQUEST,
+	PACKET_RESPONSE,
+	PACKET_UNKNOW,
+	PACKET_EVENT,
+	PACKET_DISCOVER,
+	PACKET_INFO,
+	PACKET_DISCONNECT,
+	PACKET_HEARTBEAT,
+} = require("../packets");
+
+/**
+ * Transporter for AMQP
+ *
+ * More info: https://www.amqp.org/
+ *
+ * @class AmqpTransporter
+ * @extends {Transporter}
+ */
+class AmqpTransporter extends Transporter {
+
+	/**
+	 * Creates an instance of AmqpTransporter.
+	 *
+	 * @param {any} opts
+	 *
+	 * @memberOf AmqpTransporter
+	 */
+	constructor(opts) {
+		if (typeof opts === "string")
+			opts = { amqp: { url: opts } };
+
+		// Number of requests a broker will handle concurrently
+		if (typeof opts.amqp.prefetch !== "number")
+			opts.amqp.prefetch = 1;
+
+		// Number of milliseconds before an event expires
+		if (typeof opts.amqp.eventTimeToLive !== "number")
+			opts.amqp.eventTimeToLive = 5000;
+
+		super(opts);
+
+		this.connection = null;
+		this.channel = null;
+		this.bindings = [];
+	}
+
+	/**
+	 * Connect to a AMQP server
+	 *
+	 * @memberOf AmqpTransporter
+	 */
+	connect() {
+		return new Promise((resolve, reject) => {
+			let amqp;
+			try {
+				amqp = require("amqplib");
+			} catch(err) {
+				/* istanbul ignore next */
+				this.broker.fatal("The 'amqplib' package is missing! Please install it with 'npm install amqplib --save' command!", err, true);
+			}
+
+			amqp.connect(this.opts.amqp)
+				.then(connection => {
+					this.connection = connection;
+					this.logger.info("AMQP connected!");
+
+					/* istanbul ignore next*/
+					connection
+						.on("error", (err) => {
+							this.connected = false;
+							this.logger.error("AMQP connection error!", err);
+						})
+						.on("close", (err) => {
+							this.connected = false;
+							const crashWorthy = require("amqplib/lib/connection").isFatalError(err);
+							this.logger.error("AMQP connection closed!", crashWorthy && err ||  "");
+						})
+						.on("blocked", (reason) => {
+							this.logger.warn("AMQP connection blocked!", reason);
```

Comment:
From the docs I read that a RabbitMQ server (after version 3.2.0) can decide to block a connection. 

> Typically it will do this if there is some resource shortage, e.g., memory, and messages are published on the connection. - [amqp.node](http://www.squaremobius.net/amqp.node/channel_api.html#model_events)

I don't know how common this error is, but I thought logging would help with debugging.